src/plugins/vis_augmenter/public/types.ts

```diff
@@ -56,3 +56,5 @@ export const isPointInTimeEventsVisLayer = (obj: any) => {
 export const isValidVisLayer = (obj: any) => {
   return obj?.type in VisLayerTypes;
 };
+
+export const isVisLayerWithError = (visLayer: VisLayer): boolean => visLayer.error !== undefined;
```

Comment:
`isValidVisLayer` is used for checking validity in the saved obj API that the saved obj is valid when converting into a `VisLayer`. `isVisLayerWithError` is for checking if one of the returned `VisLayer`s contains an error field. Since they serve different purposes and have different inputs, prefer to keep separate.

Reply:
Hmm i think i get it, but can you add context to both functions then? As a developer who might look at this in future, its not immediately obvious to me when I should use one over the other. Adding a comment describing the usecase for both of these functions would help :)

Reply:
Sure! I will add.